indico/modules/auth/controllers.py

```diff
@@ -200,6 +205,60 @@ def _create_user(self, data, handler):
         return handler.redirect_success()
 
 
+class RHUserAccounts(RHUserBase):
+    """Displays user accounts"""
+
+    def _create_form(self):
+        if self.user.local_identity:
+            defaults = FormDefaults(username=self.user.local_identity.identifier)
+            local_account_form = LocalLoginEditForm(identity=self.user.local_identity, obj=defaults)
+        else:
+            local_account_form = LocalLoginAddForm()
+        return local_account_form
+
+    def _handle_add_local_account(self, form):
+        identity = Identity(provider='indico', identifier=form.data['username'], password=form.data['password'])
+        self.user.identities.add(identity)
+        flash(_("Local account added successfully"), 'success')
+
+    def _handle_edit_local_account(self, form):
+        self.user.local_identity.identifier = form.data['username']
+        self.user.local_identity.password = form.data['password']
+        flash(_("Your local account credentials have been updated successfully"), 'success')
+
+    def _process(self):
+        form = self._create_form()
+        if form.validate_on_submit():
+            if isinstance(form, LocalLoginAddForm):
+                self._handle_add_local_account(form)
+            elif isinstance(form, LocalLoginEditForm):
+                self._handle_edit_local_account(form)
+            return redirect(url_for('auth.accounts'))
+        provider_titles = {name: provider.title for name, provider in multipass.auth_providers.iteritems()}
+        return WPUser.render_template('accounts.html', form=form, user=self.user, provider_titles=provider_titles)
+
+
+class RHUserAccountsRemove(RHUserBase):
+    """Removes an account"""
+
+    def _checkParams(self):
+        RHUserBase._checkParams(self)
+        self.identity = Identity.get_one(request.view_args['identity'])
+        if self.identity.user != self.user:
+            raise Forbidden("This identity doesn't belong to the user".format(self.identity.id))
```

Comment:
Missing format placeholder. Anyway, since it's a restful URL i'd even go for a plain `NotFound` (not even with a message). After all. that identity doesn't exist inside the specified user...